README.md

```diff
@@ -39,7 +39,83 @@ make test
 ./scripts/tests --help
 ```
 
+## Formal Verification
+
+We use the [C Bounded Model Checker (CBMC)](https://github.com/diffblue/cbmc) to prove absence of various classes of undefined behaviour in C, including out of bounds memory accesses and integer overflows. The proofs cover all C code in [mldsa/*](mldsa) and [mldsa/fips202/*](mldsa/fips202) involved in running mldsa-native with its C backend. See [proofs/cbmc](proofs/cbmc) for details.
+
+## Security
+
+All assembly in mldsa-native is constant-time in the sense that it is free of secret-dependent control flow, memory access,
+and instructions that are commonly variable-time, thwarting most timing side channels. C code is hardened against compiler-introduced
+timing side channels through suitable barriers and constant-time patterns.
+
+Absence of secret-dependent branches, memory-access patterns and variable-latency instructions is also tested using `valgrind`
+with various combinations of compilers and compilation options.
+
+## Design
+
+mldsa-native is split into a _frontend_ and two _backends_ for arithmetic and FIPS202 / SHA3. The frontend is
+fixed, written in C, and covers all routines that are not critical to performance. The backends are flexible, take care of
+performance-sensitive routines, and can be implemented in C or native code (assembly/intrinsics); see
+[mldsa/native/api.h](mldsa/native/api.h) for the arithmetic backend and 
+[mldsa/fips202/native/api.h](mldsa/fips202/native/api.h) for the FIPS-202 backend. mldsa-native currently
+offers backends for C, AArch64, and x86_64 - if you'd like contribute new backends, please reach out or just open a PR.
+
+## Usage
+
+Once mldsa-native reaches production readiness, you will be able to import [mldsa](mldsa) into your project's source tree and build using your preferred build system. See [mldsa](mldsa) for more information. The build system provided in this repository is for development purposes only.
+
+### Will I be able to bring my own FIPS-202?
```

Comment:
Is it really worthwhile to phrase this in future tense? We will only forget to change the phrasing later. 
I suggest we change to the phrasing used in mlkem-native.
The warning at the top that mldsa-native is not yet ready is prominent enough.

Reply:
[01f2c95](https://github.com/pq-code-package/mldsa-native/pull/385/commits/01f2c9538c8d1f2255e336efb5924ccd6cc02e63)

Reply:
Ok, added [70bfcf8](https://github.com/pq-code-package/mldsa-native/pull/385/commits/70bfcf865e0550ee7c88394df6312a44b8b8f5c1). We can increment and improve once more of the bring your own fips is complete.